compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -398,6 +404,8 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
                 // avoid type aliases for tuples
                 Right(MirrorSource.GenericTuple(types))
               case _ => reduce(tp.underlying)
+          case tp @ AppliedType(tref: TypeRef, List(_, typeTuple)) if tref == defn.NamedTupleTypeRef =>
```

Comment:
```suggestion
          case tp @ AppliedType(tref: TypeRef, List(_, _)) if tref == defn.NamedTupleTypeRef =>
```

Reply:
I ended up changing it into `defn.NamedTupleDirect(_, _)` (same AppliedType internally), for readability (I don't think that unapply was there when I started making this PR)